src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1311,7 +1341,18 @@ namespace winrt::TerminalApp::implementation
             SwitchToTabArgs switchToTabArgs{ idx.value() };
             action.Args(switchToTabArgs);
 
-            actions.push_back(action);
+            actions.emplace_back(std::move(action));
+        }
+
+        // If the user set a custom name, save it
+        if (_WindowName != L"")
+        {
+            ActionAndArgs action;
+            action.Action(ShortcutAction::RenameWindow);
+            RenameWindowArgs args{ _WindowName };
+            action.Args(args);
+
+            actions.emplace_back(std::move(action));
```

Comment:
This also raises an interesting question: what happens when the user has a persisted layout (with a window named `foo`) and does `wt -w foo nt ; sp`? 

we'll make a first window that runs `nt ; sp`, and it'll be named `foo`

the window we'll restore from he persisted layout will attempt to rename itself to `foo`, but fail because foo already exists. That seems fine to me, just something we should include in the docs/FAQsfor this.